neomodel/core.py

```diff
@@ -172,31 +179,228 @@ def refresh(self):
             raise ValueError("Can't refresh unsaved node")
 
     @classmethod
-    def create(cls, *props):
-        query = ""
-        deflated = [cls.deflate(p) for p in list(props)]
-        params = {}
-        for i in range(0, len(deflated)):
-            props = ", ".join(["{}: {{n{}_{}}}".format(key, i, key)
-                    for key, value in deflated[i].items()])
-            query += "CREATE (n{} {{{}}})\n".format(i, props)
+    def _validate_params(cls, params, required_only=False):
+        """
+        Validate property parameters for this class, split them into required and optional dicts. Raises any error if
+        properties are missing, or if optional properties were provided with required_only=True.
+
+        :param params:
+        :type params: dict
+        :param required_only: True to allow required properties only, otherwise False
+        :type required_only: bool
+        :return: A tuple of (required_params, optional_params)
+        :rtype: tuple of dict, dict
+        """
+        required_params = dict((name, None) for name in cls.__required_properties__)
+        if required_only:
+            optional_params = {}
+        else:
+            optional_params = dict(params)
+
+        # validate required params
+        for required in required_params:
+            value = params.get(required)
+            if value is None:
+                raise ValueError('Missing required property [%s] in %s' % (required, repr(params)))
+            required_params[required] = value
+            # remove required parameter
+            optional_params.pop(required, None)
+
+        return required_params, optional_params
+
+    @classmethod
+    def _build_create_query(cls, create_params):
+        """
+        Get a tuple of a CYPHER query and a params dict for the specified CREATE query.
+
+        :param create_params: The target node parameters.
+        :type create_params: dict
+        :rtype: tuple
+        """
+        props = ", ".join(["{0}: {{n_{0}}}".format(key) for key in create_params])
+        params = dict(("n_{}".format(key), value) for key, value in create_params.items())
+        query = "CREATE (n {{{}}})\n".format(props)
+        # add all inherited labels to the created entity
+        for label in cls.inherited_labels():
+            query += "SET n:`{}`\n".format(label)
```

Comment:
@Hexadite-Max think we could get rid of a slim amount of overhead by transitioning this to a list comprehension like you've done in most other areas?